### PR TITLE
93 train test split

### LINKED ISSUE #93: Function: Train Test split
Implementation in `motrainer.py`
- [ ] If specify {"coords":mask} or {"dims":mask}, split according to mask 
- [ ] Else, wrap kwargs to `dask_ml.model_selection.train_test_split`

### PR DESCRIPTION
**Pull request by rogerkuou**
fix #93

### CODE REVIEW

**Comment by rogerkuou** [comment]
Hi @SarahAlidoost, when you have time, could you please review this one for me? Thanks!

File: motrainer/splitter.py [hunk 1]
```diff
@@ -114,6 +114,61 @@ def dataset_split(ds: xr.Dataset, identifier: dict | str) -> db:
     return bags
 
 
+def train_test_split(
+    ds: xr.Dataset,
+    mask: xr.DataArray = None,
+    split: dict = None,
+    reverse: bool = False,
+) -> tuple[xr.Dataset, xr.Dataset]:
+    """Split train and test dataset.
```
Comment (SarahAlidoost) [inline motrainer/splitter.py:123]:
```suggestion
    """Split data to train and test datasets.
```

File: motrainer/splitter.py [hunk 2]
```diff
@@ -114,6 +114,61 @@ def dataset_split(ds: xr.Dataset, identifier: dict | str) -> db:
...
+    The split is performed either 1) by specifying the training data mask (`mask`)
+    where training data locations are True, or 2) by a specifying a coordinate value
+    (`split`) splitting the data into two.
+
+    Parameters
+    ----------
+    ds : xr.Dataset
+        Xarray dataset to split
+    mask : xr.DataArray, optional
+        Mask, True at training data localtions. By default None
```
Comment (SarahAlidoost) [inline motrainer/splitter.py:134]:
```suggestion
        Mask, True at training data locations. By default None
```

File: motrainer/splitter.py [hunk 3]
```diff
@@ -114,6 +114,61 @@ def dataset_split(ds: xr.Dataset, identifier: dict | str) -> db:
...
+    ------
+    ValueError
+        When neither mask nor split is specified.
+    ValueError
+        When both mask and split are specified.
+    """
+    if all([mask is None, split is None]):
+        raise ValueError("Specify either mask or split.")
+    elif all([mask is not None, split is not None]):
+        raise ValueError("Cannot specify mask and split at the same time.")
```
Comment (SarahAlidoost) [inline motrainer/splitter.py:156]:
```suggestion
    if mask is None and split is None:
        raise ValueError("Either mask or split should be specified.")
    elif mask is not None and split is not None:
        raise ValueError("Only one of mask and split should be specified.")
```

File: tests/test_splitter.py [hunk 4]
```diff
@@ -95,6 +95,38 @@ def test_split_space_no_coords(self, ds_valid):
...
+        assert test.space.shape[0] == 6
+
+    def test_masksplit_reverse(self, ds_valid):
+        train, test = train_test_split(ds_valid, split={"space": 4}, reverse=True)
+        assert train.space.shape[0] == 6
+        assert test.space.shape[0] == 4
+
+    def test_errors(self, ds_valid):
+        with pytest.raises(ValueError):
+            _ = train_test_split(ds_valid)
```
Comment (SarahAlidoost) [inline tests/test_splitter.py:123]:
```suggestion
        with pytest.raises(ValueError) as error_msg:
            _ = train_test_split(ds_valid)
            assert "Either mask or split should be specified" in str(error_msg.value)
```

File: motrainer/splitter.py [hunk 5]
```diff
@@ -114,6 +114,61 @@ def dataset_split(ds: xr.Dataset, identifier: dict | str) -> db:
...
+        _validate_train_test_split(split)
+        mask = ds[list(split.keys())[0]] < list(split.values())[0]
+
+    train = ds.where(mask, drop=True)
+    test = ds.where(~mask, drop=True)
+
+    if reverse:
+        return (test, train)
+    else:
+        return (train, test)
```
Comment (SarahAlidoost) [inline motrainer/splitter.py:169]:
```suggestion
    return (test, train)  if reverse else (train, test)
```

File: motrainer/splitter.py [hunk 6]
```diff
@@ -167,3 +222,13 @@ def _regulate_identifier(ds: xr.Dataset, identifier: dict | str) -> dict:
...
 
     return identifier
+
+
+def _validate_train_test_split(split):
+    if isinstance(split, dict):
+        if not (set(split.keys()).issubset(MOT_DIMS) and len(split.keys()) == 1):
+            raise ValueError("split should only have 1 key")
+    else:
+        raise ValueError("split should be a dict")
```
Comment (SarahAlidoost) [inline motrainer/splitter.py:232]:
```suggestion
    if not isinstance(split, dict):
        raise ValueError("split should be a dict")
        
    if not set(split.keys()).issubset(MOT_DIMS) or len(split.keys()) != 1:
            raise ValueError("split should only have 1 key")        
```

**Comment by rogerkuou** [comment]
> @rogerkuou thanks, looks good 👍 , just some refactoring suggestions. One more thing, in the exploration notebook, there is only one example with `split` argument. Can we have one example with `mask` too?

Thanks @SarahAlidoost ! I updated the exploration notebook with an mask option. Will add this to the incoming documention
